### PR TITLE
Correct compiile errors and a problem in one of the makes

### PR DESCRIPTION
**Pull request by jslevinson-cloud**
Some of the iknow files would not compile.  These compile errors were corrected.  One of the make files had a defect because an include file could not be found.  A file deleted from Perforce needed to be deleted from GitHub.

### CODE REVIEW

File: modules/core/core.mak [hunk 1]
```diff
@@ -8,7 +8,7 @@ CPP_LANGUAGE = 14
 
 STRICT = 1
 
-INCLUDEDIRS = $(ROOT_DIR)/modules/core/src/headers $(ICUDIR)/include $(ROOT_DIR)/modules/base/src/headers $(ROOT_DIR)/modules/ali $(ROOT_DIR)/shared/System/unix $(ROOT_DIR)/shared/System $(ROOT_DIR)/shared/Utility $(ROOT_DIR)/kernel/common/h
+INCLUDEDIRS = $(ROOT_DIR)/modules/shell/src $(ROOT_DIR)/modules/core/src/headers $(ICUDIR)/include $(ROOT_DIR)/modules/base/src/headers $(ROOT_DIR)/modules/ali $(ROOT_DIR)/shared/System/unix $(ROOT_DIR)/shared/System $(ROOT_DIR)/shared/Utility $(ROOT_DIR)/kernel/common/h
```
Comment (woodfinisc) [inline modules/core/core.mak]: core should not have any dependencies on shell, the relationship goes in the other direction. Why was this necessary?

Comment (jslevinson-cloud) [reply modules/core/core.mak]: I do not know why theoretically necessary, but on Ubuntu running the make without this addition there was a compile error about a missing include.  It could be that there are files in GitHub that have been deleted from Perforce causing this compile error.

Comment (woodfinisc) [reply modules/core/core.mak]: What's the error?

Comment (jslevinson-cloud) [reply modules/core/core.mak]: I did a make clean, and then reran the make after restoring core.mak to its original state.  The make succeeded.  Probably Jos corrected the problem.  So the change to core.mak can be backed off.  I will back it off after I understand how in GitHub one backs off a change to a single file.

Comment (jslevinson-cloud) [reply modules/core/core.mak]: I removed the change locally on jslevinson-cloud/iknow.  I then committed and pushed the change.  The core.mak makefile should be back to its original state post whatever changes were made in intersystems/iknow.

Comment (jslevinson-cloud) [reply modules/core/core.mak]: I wrote to you earlier that I did a make clean and then make all succeeded.  So changes subsequent to my change to core.mak must have resolved the problem.

**Comment by bdeboe** [comment]
Maybe @nwoebcke can verify if any such file deletions weren't completely effected on the public repo?
But @jslevinson-cloud is correct that these files should go, right?

**Comment by nwoebcke** [comment]
Hi @bdeboe, I believe that the changes are in @jslevinson-cloud's branch which is yet to be merged into master.

**Comment by bdeboe** [comment]
I meant they might already have been removed on the Perforce end by Jos' latest change, but for some reason not have been removed by the Git-Fusion link? Just verifying whether Git-Fusion is as complete as we hope it would be, or as cautious as it perhaps should be :-)

**Comment by bdeboe** [comment]
Hi @jslevinson-cloud: I believe the issues you submitted this PR for have now been resolved by @JosDenysGitHub's latest commits (which came in through refreshing from our pre-existing source control infrastructure. He used a slightly different fix, but the net result (things working!) should be the same.
Can you confirm this now all works for you so we can close this?

**Comment by jslevinson-cloud** [comment]
Hi Benjamin,

I did a build on a freshly cloned previously empty directory, and it built without errors.

Cheers,
Jonathan

**Comment by bdeboe** [comment]
Great, thanks for the quick confirmation!
